cirq/ops/eigen_gate.py

```diff
@@ -165,28 +160,40 @@ def _eigen_components(self) -> List[Union[EigenComponent,
         """
         pass
 
-    # virtual method
-    def _canonical_exponent_period(self) -> Optional[float]:
-        """Determines how the exponent parameter is canonicalized.
+    def _period(self) -> Optional[float]:
+        """Determines how the exponent parameter is canonicalized when equating.
 
         Returns:
             None if the exponent should not be canonicalized. Otherwise a float
             indicating the period of the exponent. If the period is p, then a
```

Comment:
a bit strange that this returns a float, when it only every returns an integer period.

Reply:
It returns non integer in some cases, there are tests that trigger this behavior.